packages/alfa-style/src/property/will-change.ts

```diff
@@ -0,0 +1,51 @@
+import { Parser } from "@siteimprove/alfa-parser";
+import {
+  CustomIdent,
+  Keyword,
+  List,
+  type Parser as CSSParser,
+} from "@siteimprove/alfa-css";
+
+import { Longhand } from "../longhand.js";
+
+const { either } = Parser;
+
+type Specified =
+  | Keyword<"auto">
+  | List<Keyword<"scroll-position"> | Keyword<"contents"> | CustomIdent>;
+type Computed = Specified;
+
+const illegalCustomIdents = [
+  "unset",
+  "initial",
+  "inherit",
+  "will-change",
+  "auto",
+  "scroll-position",
+  "contents",
+];
+
+const animatableFeature: CSSParser<
+  Keyword<"scroll-position"> | Keyword<"contents"> | CustomIdent
+> = either(
+  either(Keyword.parse("scroll-position"), Keyword.parse("contents")),
+  CustomIdent.parse(
+    (customIdent) => !illegalCustomIdents.includes(customIdent.value),
+  ),
```

Comment:
**Question:** Given that for this property, the `<custom-ident>` must be property names, should we restrict that to property names? It feels this would makes things easier at a later stage when checking if that property is defined. OTOH, we'd need to accept all the existing properties, not just the ones we do support, otherwise we'd risk discarding valid declarations 🤔

Reply:
Should we add a list of all CSS properties, which will be independent from the ones we support, to check against?

Reply:
I'm not sure of it. It will be a pretty huge list (it must also contain the shorthands). And it will not help us in typing/being sure of the `get` because the unsupported properties will still fail.

That is, if we have the list of supported properties, we can use stuff like type `Property.Name`, or value `Object.keys(Longhands)`, … and ensure that `style.computed("will-change").some(property => style.cascaded(property).isSome())` works to detect whether one of the "will change" property exists on a given element. But since we don't support all properties, we can't guarantee that `property` has type `Property.Name` and thus that the calls will type correctly, and we'll need some extra poking and safeguarding around. So the benefits of effectively restricting them is low.

OTOH, the cost is somewhat high as we need a long list that also needs to be maintained when new properties are added to CSS (or at least used in `will-change`), which is not in our hands (vs auto-magically getting the names from the properties we support).

So, I am rather leaning toward simplicity and not trying to restrict it now. We may revisit that decision later if the benefits get higher.
Also, we should see how browsers react on a `will-change` for a non-existing property. Notably, do they discard the full declaration as invalid; or only ignore the unknown property? Due to forward-compatibility requirements that is often baked in CSS, I wouldn't be surprised of the later (to ensure old browsers don't totally break on newer pages). In that case, we must definitely not restrict ourselves. OTOH, if they do discard the full declaration, then we probably should do so (or we'll parse incorrect declaration and do the wrong thing in these cases) (but may decide not to for the sake of simplicity and revisit once we actually encounter a messed up page…)

Reply:
I did a quick test in Chrome: `will-change: scroll-position, foo` computed to `will-change: scroll-position` (also tried `will-change: foo` which expectedly computed to `will-change: auto`).